### ISSUE TITLE
expo-analytics-segment is not pushing session id to Amplitude destination(It is always tracked as -1)

### ISSUE THREAD

**Issue by codehippie1** [expo/expo#14581]
### Summary

I am using expo-analytics-segment(v 10.2.1) to send tracking info to Amplitude(Configured as the destination in app.segment.com) in an expo react native app. Though I am sending session info(epoch time) - The session always gets registered as -1.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

42

### Environment

Expo CLI 4.12.0 environment info:
    System:
      OS: Windows 10 10.0.19042
    Binaries:
      Node: 14.4.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Users\anees\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.5 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6514223
    npmPackages:
      expo: ^42.0.0 => 42.0.3
      react: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

Sample code `App.tsx`

```
Segment.initialize({
  androidWriteKey: 'androidKey', // from Segment
  iosWriteKey: 'iOsKey', // from segment
});

global.epochInMilliSeconds = Date.now();
Segment.identifyWithTraits(
  user.sub,
  { email: 'notgood@gmail.com' },
  {
    event: 'App Started',
    integrations: {
      Amplitude: {
        sessionId: global.epochInMilliSeconds,
      },
    },
  }
);

Segment.trackWithProperties(
  'App Started',
  { email: 'fancyemail@gmail.com' },
  { integrations: { Amplitude: { session_id: global.epochInMilliSeconds } } }
); <------------------- Did not work. Session id is -1**

Segment.track('App Started'); // <-----------------------Session id is -1
```

Followed this article to spend epoch time
https://community.amplitude.com/instrumentation-and-data-management-57/how-do-we-set-session-in-amplitude-while-using-segment-in-cloud-mode-111

**Comment by cruzach** [comment]
FYI Segment has recently released a brand new react native library that is Expo-compatible, and it is likely that we will deprecate `expo-analytics-segment` in favor of their new library. 

I recommend trying out this new library if you're looking for the best Segment experience: https://github.com/segmentio/analytics-react-native/blob/master/README.md

**Comment by codehippie1** [comment]
@cruzach This is good news. Unfortunately - the new library is not production ready(My app is already in production). Nevertheless - Thank goodness for this info, will eventually move out from ill documented, not working expo-analytics-segment

**Comment by jobondi** [comment]
> @jobondi This is interesting. In my case even though segment is receiving the session id in iOS(in Android - integrations object is empty in segment itself - so no hope there), it does not seem to end up on amplitude.
> 
> See same events **Segment**
> 
> ```
>     "Amplitude": {
>       "sessionId": 1635037849037
>     }
> ```
> 
> **amplitude**
> 
> ```
>   "session_id": -1,
>   "start_version": "2.0.0",
> ```

We are seeing this same issue @codehippie1, where the integrations {} object passed in via the `options` parameter is not appearing in the Segment source debugger data, which makes me assume it's not being sent up by the SDK.

**Comment by jobondi** [comment]
Thanks for the update @cruzach , we will put in our roadmap to transition to the `@segment/analytics-react-native` library that is maintained by Segment. As @codehippie1 mentioned their library is not yet production-ready and also it looks like it does not support the Expo managed workflow, so we will also have to plan to migrate our production app to EAS builds, which is also not yet production ready (is that a correct assumption too?)  What would be the best way to make or get an updated build of expo-analytics-segment that contains the latest stable segment SDK's for iOS and Android? It seems that would resolve these little remaining issues and provide a ample time to transition a production app to both EAS build as well as the `@segment/analytics-react-native` library.

**Comment by cruzach** [comment]
you should be able to use the Expo Go client and managed workflow, there's only one native method in the library and it's not required, it just gets device info which is all doable from expo-device anyways

**Comment by jobondi** [comment]
Thanks @cruzach, that's good to know about EAS Build and we can plan to get it on our roadmap to migrate at some point soon.

The [Segment online docs for the React Native SDK](https://segment.com/docs/connections/sources/catalog/libraries/mobile/react-native/) does not mention installation steps for Expo managed workflow. Is there a doc or example or something referencale you know of that could show how we could use this and inject the info from expo-device?

It sounds like these bugs for `expo-analytics-segment` will be left as-is if it will be deprecated in favor of the `@segment/analytics-react-native`, or is there any path to keeping the `expo-analytics-segment` as the simple interface and installation and then having that be backed by the `@segment/analytics-react-native` library?

**Comment by cruzach** [comment]
just run `yarn add @segment/analytics-react-native`, you install the package as you would any other pure JS library

**Comment by codehippie1** [comment]
@jobondi @croaky @cruzach Guys - Have to come back here again with the experience with `@segment/analytics-react-native` library. 

**Short story** - It won't work with expo managed workflow(SDK 43) - just because of the dependency to `AsyncStorage`.  It produces the infamous `[@RNC/AsyncStorage]: NativeModule: AsyncStorage is null` issue while running.
**Long Story** - react team moved `AsyncStorage` to `react-community`, then again to another github repo. Expo SDK 43 still relies on `AsyncStorage` from `react-native` repo. Segment internally uses the `react-native-community` one. The latest one is somewhere else. God know where else it would be in a month.

I would gladly lived in a react-native expo managed workflow world where people doesn't fancy moving libraries around for no apparent benefits.
@cruzach When would I be able to see a pre-release of an expo sdk with latest `AsyncStorage`

**Comment by jobondi** [comment]
@cruzach we migrated from Expo managed workflow to EAS build, as well as Expo SDK 43 in the process. It looks like the yard add pulls in @segment/analytics-react-native version 1.5.0 by default. The readme says to rin the cocoapods commands but I don't recall needing to do that for any other libraries (maybe those steps are only for bare workflow / old "detached" React Native projects?)

Either way, here is the error when the app attempts to start:

"Error: Failed to load Analytics native module."

Is there an entry to the app.config.ts file needed (we needed this for another React Native library, we had to specify something in the "plugins" section)

I will check out the 2.0 though that looks like it still has a -pilot suffix on it and very small amount of weekly downloads on the npm. Is this transitioning to GA anytime in the near future?
